include/fiction/io/print_layout.hpp

```diff
@@ -307,6 +317,110 @@ void print_cell_level_layout(std::ostream& os, const Lyt& layout, const bool io_
     // flush stream
     os << std::endl;
 }
+/**
+ * Writes a simplified 2D representation of an SiDB charge layout to an output stream.
+ *
+ * @tparam Lyt Cell-level layout based on SiQAD coordinates.
+ * @param os Output stream to write into.
+ * @param cds The charge distribution surface of which the charge layout is to be printed.
+ * @param cs_color Flag to utilize color escapes for charge states.
+ */
+template <typename Lyt>
+void print_charge_layout(std::ostream& os, const charge_distribution_surface<Lyt>& cds,
+                         const bool cs_color = true) noexcept
+{
+    // empty layout
+    if (cds.num_cells() == 0ul)
+    {
+        os << "[i] empty layout" << std::endl;
+        return;
+    }
+
+    std::vector<std::pair<uint64_t, std::pair<double, double>>> sorted_locs;
+
+    auto min_x_d = std::numeric_limits<double>::max();
+    auto max_x_d = std::numeric_limits<double>::min();
+
+    for (uint64_t i = 0; i < cds.num_cells(); ++i)
+    {
+        min_x_d = std::min(min_x_d, cds.get_all_sidb_locations_in_nm()[i].first);
+        max_x_d = std::max(max_x_d, cds.get_all_sidb_locations_in_nm()[i].first);
+
+        sorted_locs.push_back(std::make_pair(i, cds.get_all_sidb_locations_in_nm()[i]));
+    };
```

Comment:
I think you could use `cds.foreach_cell` here and then simply add each cell to a vector.

Reply:
I implemented it in a new commit